### PR TITLE
Weaviate & Supabase support

### PR DESCRIPTION
**Pull request by Elliot-Construct**
Added weaviate support, currently untested. 
Suggest creating new branch for this P/R for testing purposes.

### CODE REVIEW

**Comment by homanp** [comment]
@Elliot-Construct this looks amazing, I’ll test it and merge!

File: app/lib/vectorstores/weaviate.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+           index_name=weaviate_index,
+           namespace=namespace,
+           client=weaviate_client,
+        )
+
+    def from_existing_index(self, embeddings, index_name):
+         return Weaviate.from_existing_index(
+            index_name=weaviate_index,
+            embedding=embeddings,
+            client=weaviate_client
```
Comment (homanp) [inline app/lib/vectorstores/weaviate.py:30]: This should include the namespace for filtering as as well. Otherwise it will query all docs in the index (see Pinecone example).

File: app/lib/vectorstores/supabase.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+supabase_table = "superagent"
+
+class SupabaseVectorStore:
+
+    def __init__(self):
+        pass
+
+    def from_documents(self, docs, embeddings, table_name):
+        return SupabaseVectorStore.from_texts(
+            texts=docs,
```
Comment (homanp) [inline app/lib/vectorstores/supabase.py:19]: Is there a way to filter using metadata such as the `namespace` in Pinecone and Weaviate?

**Review by homanp — CHANGES_REQUESTED** [review]
Some small comments @Elliot-Construct

File: app/lib/vectorstores/weaviate.py (hunk 1 above)
Comment (Elliot-Construct) [reply app/lib/vectorstores/weaviate.py:31]: Done this

File: app/lib/vectorstores/supabase.py (hunk 2 above)
Comment (Elliot-Construct) [reply app/lib/vectorstores/supabase.py:19]: I believe we may be looking for 'Collections' in supabase?
What do you think?

Comment (Elliot-Construct) [reply app/lib/vectorstores/supabase.py:19]: Collections and namespaces are generally different concepts:

A namespace (in Pinecone/Weaviate) is a way to segregate or filter vectors stored in the same underlying index. It's an attribute used during query time.
A collection typically refers to a separate index or database table. Storing vectors in different collections means they are in entirely separate indexes/tables.
Supabase is a relational database, so it doesn't have a direct notion of namespaces like Pinecone/Weaviate.

However, Supabase allows filtering by metadata columns during query time. This can function similarly to namespaces - you store a "namespace" column with each vector and filter by it when querying.

So in summary:

Pinecone/Weaviate namespaces: Filtering within the same index
Supabase "namespaces": Filtering by metadata column
Different collections: Entirely separate indexes/tables
Since Supabase doesn't directly support namespaces, using a metadata column to filter results is the closest equivalent.

So it might be wise to implement a namespace column in the metadata in the collection creation process.

**Review by homanp — CHANGES_REQUESTED** [review]
Can't seem to get this to work properly with weaviate. Would you mind installing and running this tinder if you can fix it? @Elliot-Construct
